packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -269,4 +297,62 @@ export class SCWSigner implements Signer {
     }
     return true;
   }
+
+  private async addAddress(request: RequestArguments) {
+    const state = SubAccount.getState();
+    if (state.account) {
+      return state.account;
+    }
+
+    await this.communicator.waitForPopupLoaded?.();
+    let signer = get(request, 'params[0].signer') as string;
+    if (!state.getSigner) {
+      throw standardErrors.rpc.invalidParams('no signer provided');
+    }
+    const account = await state.getSigner();
+    if (!signer && account) {
+      signer = account.publicKey;
+    }
+    if (!Array.isArray(request.params)) {
+      throw standardErrors.rpc.invalidParams();
+    }
+    request.params[0].signer = signer;
+
+    const response = await this.sendRequestToPopup(request);
+
+    // Only store the sub account information after the popup has been closed and the
+    // user has confirmed the creation
+    SubAccount.setState({
+      account: response as SubAccountInfo,
```

Comment:
For these, it also would be nice have assertion functions over casting.

Reply:
+1